lib/open_meteo/entities/forecast.rb

```diff
@@ -9,29 +10,27 @@ module Entities
     #
     # A forecast Entity with data returned by OpenMeteo
     class Forecast
-      attr_reader :attributes, :current, :hourly, :daily, :raw_json
+      attr_reader :attributes, :current, :minutely_15, :hourly, :daily, :raw_json
 
       def initialize(json_body)
         @raw_json = json_body
 
-        @current =
-          json_body["current"] &&
-            OpenMeteo::Entities::Forecast::Current.new(
-              json_body["current"],
-              json_body["current_units"],
-            )
-        @hourly =
-          json_body["hourly"] &&
-            OpenMeteo::Entities::Forecast::Hourly.new(
-              json_body["hourly"],
-              json_body["hourly_units"],
-            )
-        @daily =
-          json_body["daily"] &&
-            OpenMeteo::Entities::Forecast::Daily.new(json_body["daily"], json_body["daily_units"])
+        set_data(json_body, :current, OpenMeteo::Entities::Forecast::Current)
+        set_data(json_body, :minutely_15, OpenMeteo::Entities::Forecast::Minutely15)
+        set_data(json_body, :hourly, OpenMeteo::Entities::Forecast::Hourly)
+        set_data(json_body, :daily, OpenMeteo::Entities::Forecast::Daily)
 
         @attributes = json_body.keys
       end
+
+      def set_data(json_body, type, klass)
+        return if json_body[type.to_s].nil?
+
+        instance_variable_set(
+          "@#{type}",
+          klass.new(json_body[type.to_s], json_body["#{type}_units"]),
```

Comment:
The `type` could have been also deduced from klass.

```ruby
klass.to_s.split("::").last.underscore.to_sym
```

Reply:
Good idea 👌😊

Reply:
But maybe with minute 15 would need to be adjusted...